libusb.go

```diff
@@ -301,12 +301,14 @@ func (libusbImpl) getDeviceDesc(d *libusbDevice) (*DeviceDesc, error) {
 			}
 			descs := make([]InterfaceSetting, 0, len(alts))
 			for _, alt := range alts {
+
 				i := InterfaceSetting{
 					Number:     int(alt.bInterfaceNumber),
 					Alternate:  int(alt.bAlternateSetting),
 					Class:      Class(alt.bInterfaceClass),
 					SubClass:   Class(alt.bInterfaceSubClass),
 					Protocol:   Protocol(alt.bInterfaceProtocol),
+					Extra:      C.GoBytes(unsafe.Pointer(alt.extra), alt.extra_length),
```

Comment:
I don't think this works correctly. The buffer that alt.extra points to is owned by libusb and will be freed the moment free_config_descriptor is called (line 342 below). I think to make it work, you would need to make copies of the underlying buffers. But also see my comment above: I think this format is insufficient and you'll need to add more logic here. The good news is that the same logic can be reused for the unsupported descriptor processing on all descriptor levels.

Reply:
Doesn't `GoBytes` make a copy already? [a bit superficial note on the docs...](https://pkg.go.dev/cmd/cgo#:~:text=A%20few%20special%20functions%20convert%20between%20Go%20and%20C%20types%20by%20making%20copies%20of%20the%20data.%20In%20pseudo%2DGo%20definitions%3A)

Reply:
oh, I think you're right, I missed that. Thanks!